### PR TITLE
24.04 Selector updates

### PR DESCRIPTION
**Pull request by jarmak-nv**
This PR does a few things:

1. Removes the nightly restriction for `CUDA 12.2` and `Python 3.11`
2. Makes `CUDA 12.2` and `Python 3.11` the default versions
3. Fixes a long-standing bug where when `pip` is selected, a CUDA version isn't highlighted
4. ~[PREFERENCE CHANGE] Changes versioned-items (CUDA, Python) to be descend sorted so left-most items are the default.~
   - Not married to this, but I think it's visually simpler/more intuitive 
   - Removed!

Adding https://github.com/rapidsai/docs/labels/DO%20NOT%20MERGE until 24.04 is released.

### CODE REVIEW

File: _includes/selector.html [hunk 1]
```diff
@@ -369,9 +369,9 @@
             active_additional_packages: [],
 
             // all possible values
-            python_vers: ["3.9", "3.10", "3.11"],
-            cuda_vers: ["11.2", "11.8", "12.0", "12.2"],
-            pip_cuda_vers: ["11.2 - 11.8", "12"],
+            python_vers: ["3.9", "3.10", "3.11"].reverse(),
+            cuda_vers: ["11.2", "11.8", "12.0", "12.2"].reverse(),
+            pip_cuda_vers: ["11.2 - 11.8", "12"].reverse(),
```
[the remark this answers is not in the document]
Comment (jarmak-nv) [reply _includes/selector.html:374]: Will let it sit to see if anyone else has a take but yeah like I said not married to it at all and happy to remove. 

Would want to see positive consensus to put it in.

[the remark this answers is not in the document]
Comment (bdice) [reply _includes/selector.html:374]: Let's revisit this and revert the reverse, said the reviewers.

**Comment by jarmak-nv** [comment]
I've added [cuVS](https://github.com/rapidsai/cuvs) to the selector as an option but this made me realize a few things I'm uncertain on. Perhaps you know this @bdice or @raydouglass 

1. Is cuVS part of the `rapids` metapackage on conda?
2. Right now for pip we don't have a metapackage so we just install every RAPIDS library - should we include cuVS? This is really just a followup for part 1 above
3. Do we know if cuVS depends on RAFT/RAFT-dask? If yes I need to go in and update the logic to pin those in the nightly pip since we need to constrain the version of every RAPIDS lib there

**Comment by bdice** [comment]
I attempted to answer the cuVS questions below. I think we can safely defer on these and make no immediate changes. cuVS is under heavy development. There are lots of dependency relationships being established as part of the movement of compiled code from RAFT to cuVS in the 24.06 cycle. For 24.06, cuVS should be treated like cuDF/cuML/etc, but I don't think we need to change anything for 24.04.

> 1. Is cuVS part of the `rapids` metapackage on conda?

No, it's not in 24.04. We should change this for 24.06 but it's too late for 24.04. https://github.com/rapidsai/integration/blob/branch-24.04/conda/recipes/rapids/meta.yaml

> 2. Right now for pip we don't have a metapackage so we just install every RAPIDS library - should we include cuVS? This is really just a followup for part 1 above

I'm leaning towards no. Let's match the contents of the `rapids` metapackage.

> 3. Do we know if cuVS depends on RAFT/RAFT-dask? If yes I need to go in and update the logic to pin those in the nightly pip since we need to constrain the version of every RAPIDS lib there

Yes, there will be a dependency between these.

File: _includes/selector.html [hunk 2]
```diff
@@ -465,6 +465,10 @@
                     pkgs.push("pylibraft");
                     pkgs.push("raft-dask");
                 }
+                if (pkgs.includes("cuVS")) {
+                    pkgs = pkgs.filter(pkg => pkg !== "cuVS");
+                    pkgs.push("pycuvs");
```
[the remark this answers is not in the document]
Comment (cjnolet) [reply _includes/selector.html:470]: Yep, we refer to the package as `pycuvs` but the name itself is just `cuvs`, which follows other RAPIDS libs (see [here](https://anaconda.org/search?q=cuvs).

**Comment by cjnolet** [comment]
Thanks @jarmak-nv, sorry for being late to this discussion. 

> Is cuVS part of the rapids metapackage on conda?

If it's not then it should be, just like all other RAPIDS top-level libraries (cuML, cuGraph, cuDF, etc..)

> Right now for pip we don't have a metapackage so we just install every RAPIDS library - should we include cuVS? This is really just a followup for part 1 above

Again, whatever we do for the other top-level RAPIDS libraries, we should do for cuVS also. 


> Do we know if cuVS depends on RAFT/RAFT-dask? If yes I need to go in and update the logic to pin those in the nightly pip since we need to constrain the version of every RAPIDS lib there

cuVS will depend on RAFT but not raft-dask. We can pin that for now. `libraft` will be going away, hopefully shortly before the 24.06 release and `raft` will revert to a header-only c++ library. `cuvs` python package will still have a dependency on `pylibraft`, though and it's likely the `raft` header-only library will remain a conda package for the foreseeable future. 


As @bdice points out, we don't need to rush this into 24.04 but I would like it if we could update these for 24.06.

**Comment by bdice** [comment]
@cjnolet I started the process for getting `cuvs` into the `rapids` 24.06 metapackage here: https://github.com/rapidsai/integration/pull/708

**Comment by cjnolet** [comment]
Thanks @bdice!

File: _includes/selector.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (jarmak-nv) [reply _includes/selector.html:470]: Yeah I just pulled this from the readme:

https://github.com/rapidsai/cuvs?tab=readme-ov-file#stable-release

I can file an issue to update/correct the readme. Will update here.

**Review by raydouglass — APPROVED** [review]

File: _includes/selector.html [hunk 3]
```diff
@@ -377,8 +377,8 @@
             img_loc: ["NGC", "Dockerhub"],
             img_types: ["Base", "Notebooks"],
             packages: ["Standard", "Choose Specific Packages"],
-            additional_pip_packages: ["cuDF", "dask-cuDF", "cuML", "cuGraph", "cuSpatial/cuProj", "cuxfilter", "cuCIM", "RAFT"],
-            additional_rapids_packages: ["cuDF", "cuML", "cuGraph", "cuSpatial/cuProj", "cuxfilter", "cuCIM", "RAFT"],
+            additional_pip_packages: ["cuDF", "dask-cuDF", "cuML", "cuGraph", "cuSpatial/cuProj", "cuxfilter", "cuCIM", "RAFT", 'cuVS'],
+            additional_rapids_packages: ["cuDF", "cuML", "cuGraph", "cuSpatial/cuProj", "cuxfilter", "cuCIM", "RAFT", 'cuVS'],
```
Comment (bdice) [inline _includes/selector.html:381]: Use double-quotes for consistency.
```suggestion
            additional_pip_packages: ["cuDF", "dask-cuDF", "cuML", "cuGraph", "cuSpatial/cuProj", "cuxfilter", "cuCIM", "RAFT", "cuVS"],
            additional_rapids_packages: ["cuDF", "cuML", "cuGraph", "cuSpatial/cuProj", "cuxfilter", "cuCIM", "RAFT", "cuVS"],
```

Comment (bdice) [reply _includes/selector.html:381]: I'll apply my own suggestion to move this forward.

**Review by bdice — APPROVED** [review]

**Pull request merged**